x/datapool/handler.go

```diff
@@ -24,6 +24,12 @@ func NewHandler(k keeper.Keeper) sdk.Handler {
 		case *types.MsgCreatePool:
 			res, err := msgServer.CreatePool(sdk.WrapSDKContext(ctx), msg)
 			return sdk.WrapServiceResult(ctx, res, err)
+		case *types.MsgRegisterNFTContract:
+			res, err := msgServer.RegisterNFTContract(sdk.WrapSDKContext(ctx), msg)
+			return sdk.WrapServiceResult(ctx, res, err)
+		case *types.MsgUpgradeNFTContract:
```

Comment:
I think this above issue is the same goes for upgrades.
Because if a malicious person deploys and migrates the forged contract, all NFTs can be hijacked.